internal/controllers/machineset/machineset_controller_status_test.go

```diff
@@ -753,6 +776,135 @@ func Test_setMachinesUpToDateCondition(t *testing.T) {
+			machines:   nil,
+			getAndAdoptMachinesForMachineSetSucceeded: false,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetRemediatingV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  clusterv1.MachineSetRemediatingInternalErrorV1Beta2Reason,
+				Message: "Please check controller logs for errors",
+			},
+		},
+		{
+			name:       "Without unhealthy machines",
+			machineSet: &clusterv1.MachineSet{},
+			machines: []*clusterv1.Machine{
+				fakeMachine("m1"),
+				fakeMachine("m2"),
+			},
+			getAndAdoptMachinesForMachineSetSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineSetRemediatingV1Beta2Condition,
+				Status: metav1.ConditionFalse,
+				Reason: clusterv1.MachineSetNotRemediatingV1Beta2Reason,
+			},
+		},
+		{
+			name:       "With machines to be remediated by MS",
+			machineSet: &clusterv1.MachineSet{},
+			machines: []*clusterv1.Machine{
+				fakeMachine("m1", withConditions(healthCheckSucceeded)),    // Healthy machine
+				fakeMachine("m2", withConditions(healthCheckNotSucceeded)), // Unhealthy machine, not yet marked for remediation
+				fakeMachine("m3", withConditions(healthCheckNotSucceeded, ownerRemediated), withV1Beta2Condition(ownerRemediatedV1Beta2)),
+			},
+			getAndAdoptMachinesForMachineSetSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetRemediatingV1Beta2Condition,
+				Status:  metav1.ConditionTrue,
+				Reason:  clusterv1.MachineSetRemediatingV1Beta2Reason,
+				Message: " from Machine m3", // FIXME: We should change something :)
+			},
+		},
+		{
+			name:       "With machines to be remediated by MS and preflight check error",
+			machineSet: &clusterv1.MachineSet{},
+			machines: []*clusterv1.Machine{
+				fakeMachine("m1", withConditions(healthCheckSucceeded)),    // Healthy machine
+				fakeMachine("m2", withConditions(healthCheckNotSucceeded)), // Unhealthy machine, not yet marked for remediation
+				fakeMachine("m3", withConditions(healthCheckNotSucceeded, ownerRemediated), withV1Beta2Condition(ownerRemediatedV1Beta2)),
+				fakeMachine("m4", withConditions(healthCheckNotSucceeded, ownerRemediated), withV1Beta2Condition(ownerRemediatedWaitingForRemediationV1Beta2)),
+			},
+			getAndAdoptMachinesForMachineSetSucceeded: true,
+			// This preflight check error can happen when a Machine becomes unhealthy while the control plane is upgrading.
+			remediationPreflightCheckErrMessage: "KubeadmControlPlane ns1/cp1 is upgrading (\"ControlPlaneIsStable\" preflight check failed)",
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineSetRemediatingV1Beta2Condition,
+				Status: metav1.ConditionTrue,
+				Reason: clusterv1.MachineSetRemediatingV1Beta2Reason,
+				Message: "Remediation is blocked because KubeadmControlPlane ns1/cp1 is upgrading (\"ControlPlaneIsStable\" preflight check failed);" +
+					"  from Machines m3, m4", // FIXME: We should change something :)
```

Comment:
@fabriziopandini Please double check if I"m right, but it looks like when we use realistic examples for Remediating conditions the resulting aggregation is not good (at least for cases where we don't have a message).

Let's discuss how we want to fix this (I think either we should add messages to the Remediating conditions on the Machines or we somehow have to aggregate based on reasons)